src/main/java/com/cloudbees/jenkins/plugins/bitbucket/filesystem/BitbucketSCMFileSystem.java

```diff
@@ -131,6 +132,8 @@ public SCMFileSystem build(@NonNull SCMSource source, @NonNull SCMHead head, @Ch
             String ref;
             if (head instanceof BranchSCMHead) {
                 ref = head.getName();
+            } else if (head instanceof TagSCMHead) {
+                ref = head.getName();
```

Comment:
May need additional changes here if there is a branch with the same name as a tag and the source is only selecting tags (so that the tag would win during fetch)

We may need to change `ref` from being just the name to being a more fully specified ref that includes the prefix from which the ref is to be retrieved (so that this would become `ref = "tags/" + head.getName()` and similar corresponding changes made elsewhere... but I haven't investigated this)